### DISCUSSION TITLE
Why my app need Reosetta 2 to work

### DISCUSSION THREAD

**Discussion by zabdoul60460** [airsdk/Adobe-Runtime-Support#1889] [General]
Hi,
I package a Mac OS release (native installer) with Visual Studio Code and AS3Mxml.
Do you know why this application works only if 'Open with Rosetta' box is checked ?
Thanks for helping.
Best regards

**Comment by ajwfrost** [comment]
Hi

Based on your other question .. if you're using HTMLLoader or the 'built-in' webview component, these are using the WebKit library which was customised by Adobe and comes with the AIR SDK .. but is only available as an x86_64 library. So if you're using M1 hardware, the AIR application would be opened up normally using the ARM64 binary, but as soon as it then tried to load the WebKit.dylib file, it will fail as all binaries in the process need to be the same architecture.
If you set it to open with Rosetta, then the x86_64 version of the AIR binaries will be run to start with, and it will then successfully load the WebKit.dylib file.

This issue is one of the reasons that we're getting rid of WebKit .. (that and the fact that it's over 10 years old now) - so the preference is to use the native capabilities available via StageWebView and you then don't have any restrictions on what binaries/silicon it runs on..

thanks
